xml_converter/src/xml_converter.cpp

```diff
@@ -62,40 +62,62 @@ vector<string> get_files_by_suffix(string directory, string suffix) {
     return files;
 }
 
-void read_taco_directory(
+map<string, vector<string>> read_taco_directory(
     string input_path,
     map<string, Category>* marker_categories,
     vector<Parseable*>* parsed_pois) {
+    map<string, vector<string>> category_to_filepath;
     if (!filesystem::exists(input_path)) {
         cout << "Error: " << input_path << " is not an existing directory or file" << endl;
     }
     else if (filesystem::is_directory(input_path)) {
+        string directory_name = filesystem::path(input_path).filename();
         vector<string> xml_files = get_files_by_suffix(input_path, ".xml");
         for (const string& path : xml_files) {
-            parse_xml_file(path, input_path, marker_categories, parsed_pois);
+            set<string> category_names = (parse_xml_file(path, input_path, marker_categories, parsed_pois));
```

Comment:
```suggestion
            set<string> top_level_category_names = parse_xml_file(path, input_path, marker_categories, parsed_pois);
```
nit: to be more specific on what category names we are getting. Also removing the extra parens.